packages/eslint-plugin-template/src/rules/no-restricted-syntax.ts

```diff
@@ -0,0 +1,96 @@
+import type { AST } from '@angular/compiler';
+import { AbsoluteSourceSpan } from '@angular/compiler';
+import type { TSESLint } from '@typescript-eslint/experimental-utils';
+import {
+  createESLintRule,
+  getTemplateParserServices,
+} from '../utils/create-eslint-rule';
+import { getLocFromAbsoluteSourceSpan } from '../utils/get-loc-from-absolute-source-span';
+
+type Options = [
+  string | { readonly message?: string; readonly selector: string },
+];
+export type MessageIds = 'noRestrictedSyntax';
+export const RULE_NAME = 'no-restricted-syntax';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallows specified syntax.',
+      category: 'Stylistic Issues',
+      recommended: false,
+    },
+    schema: {
+      type: 'array',
+      items: {
+        oneOf: [
+          {
+            minLength: 1,
+            type: 'string',
+          },
+          {
+            type: 'object',
+            properties: {
+              message: { type: 'string' },
+              selector: { type: 'string' },
+            },
+            required: ['selector'],
+            additionalProperties: false,
+          },
+        ],
+      },
+      uniqueItems: true,
+      minItems: 1,
+    },
+    messages: {
+      noRestrictedSyntax: '{{message}}',
+    },
+  },
+  defaultOptions: [''],
+  create(context, options) {
+    const sourceCode = context.getSourceCode();
+    const parserServices = getTemplateParserServices(context);
+
+    function visitor(message: string) {
+      return ({ sourceSpan }: AST) => {
+        const loc =
+          sourceSpan instanceof AbsoluteSourceSpan
+            ? getLocFromAbsoluteSourceSpan(sourceCode, sourceSpan)
+            : parserServices.convertNodeSourceSpanToLoc(sourceSpan);
+        context.report({
+          loc,
+          messageId: 'noRestrictedSyntax',
+          data: {
+            message,
+          },
+        });
+      };
+    }
+
+    return options.reduce<Record<string, TSESLint.RuleFunction>>(
+      (selectors, selectorOrOption) => {
+        const { message, selector } = getNormalizedOptions(selectorOrOption);
+        return {
+          ...selectors,
+          // If the `selector` is already computed (it can happens in case of passing `string` + object format), we just ignore it.
+          ...(!(selector in selectors) && { [selector]: visitor(message) }),
```

Comment:
To be clear, it's preventing the following case:

```ts
{
  "@angular-eslint/template/no-restricted-syntax": [
    "error",
    "BoundAttribute",
    {
      "selector": "BoundAttribute" // note that is a duplicated.
    },
  ]  
}
```

AFAIK, we can't prevent this using `schema`.

---

That being said:

1. Not sure if this comment is clear enough, let me know in case of rephrase needed.
2. Or perhaps would it be better to give a warning?

Reply:
This seems to be the opposite behavior to the existing core rule for JS: https://github.com/eslint/eslint/blob/d36f12f71b3e4f9e9552f1054d7a75be4dc03671/lib/rules/no-restricted-syntax.js#L58

The latest definition for that selector will replace previous ones via the Object.assign.

I think it would be better to match the existing rule here, regardless of whether we agree with it or not